src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -176,6 +179,9 @@ internal sealed class RSession : IRSession, IRCallbacks {
 
             var initializationTask = _initializationTcs.Task;
             _hostRunTask = CreateAndRunHost(startupInfo, timeout);
+
+            Interlocked.Exchange(ref _afterHostStartedTcs, new TaskCompletionSource<object>());
```

Comment:
There is a race between this and `await` on the task - if you call `EvaluateAsync` before this line runs, it'll be waiting on the old task (which, if this session has never been started before, will never be completed).

I think this rather needs to follow the same pattern as `EnsureHostStartedAsync` does for `_initializationTcs`.